src/routes/skills/edit/id.tsx

```diff
@@ -0,0 +1,191 @@
+import { Button } from '@/components/ui/button'
+import { Calendar } from '@/components/ui/calendar'
+import { Checkbox } from '@/components/ui/checkbox'
+import { FormControl, FormField, FormItem, FormLabel, FormMessage, Form as FormProvider } from '@/components/ui/form'
+import { Input } from '@/components/ui/input'
+import { Popover, PopoverContent, PopoverTrigger } from '@/components/ui/popover'
+import { Select, SelectContent, SelectItem, SelectTrigger } from '@/components/ui/select'
+import { useToast } from '@/components/ui/use-toast'
+import { config } from '@/config'
+import { cn } from '@/lib/utils'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { SelectValue } from '@radix-ui/react-select'
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query'
+import { useNavigate, useParams } from '@tanstack/router'
+import { format } from 'date-fns'
+import { Calendar as CalendarIcon } from 'lucide-react'
+import { useForm } from 'react-hook-form'
+import z from 'zod'
+const formSchema = z.object({
+  name: z.string().nonempty({ message: 'Name is required' }),
+  proficiency: z.string().regex(/^(100(\.0{1,2})?|\d{1,2}(\.\d{1,2})?)%$/, { message: 'Invalid proficiency' }),
+  logo: z.string({ required_error: 'test' }).url({ message: 'Invalid URL' })
+})
+
+type Form = z.infer<typeof formSchema>
+
+type Response =
+  | {
+      name: string
+      proficiency: string
+      logo: string
+    }
+  | { message: string }
+
+export function EditSkill() {
+  const navigate = useNavigate()
+  const queryClient = useQueryClient()
+  const { id } = useParams()
+  const { data: skill, isLoading } = useQuery({
+    queryKey: ['skill', id],
+    queryFn: async () => {
+      return fetch(`${config.VITE_BACKEND_URL}/resume/skill/${id as string}`, {
+        method: 'GET'
+      }).then((res) => res.json() as Promise<Response>)
+    },
+    enabled: !!id
+  })
+  const { toast } = useToast()
+  const addSkill = useMutation({
+    mutationFn: (data: Required<Form>) => {
+      return fetch(`${config.VITE_BACKEND_URL}/resume/skill/${id as string}`, {
+        body: JSON.stringify({
+          ...data
+        }),
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json'
+        }
+      })
+      
+      return fetch(`${config.VITE_BACKEND_URL}/resume/skill/${id as string}`, {
```

Comment:
I think this is a duplicate put request

Reply:
Thanks for catching that. I've deleted it.